src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -118,6 +121,26 @@ public virtual void Start()
             _hostedServiceExecutor.Start();
 
             _logger.Started();
+
+            // REVIEW: Is this the right place to log these errors?
+            if (_hostingStartupErrors != null)
+            {
+                foreach (var exception in _hostingStartupErrors.InnerExceptions)
+                {
+                    _logger.HostingStartupAssemblyError(exception);
+                }
+            }
+            else
+            {
+                // If there were no errors then just log the fact that we did load hosting startup assemblies.
+                if (_logger.IsEnabled(LogLevel.Debug))
+                {
+                    foreach (var assembly in _options.HostingStartupAssemblies)
```

Comment:
If one fails and one succeeds then the success does not get logged. How about _hostingStartupErrors becomes _hostingStartupResults, `IEnumerable<KeyValuePair<string, exception>>` for the assembly name and optional error.